integration/test/Test/LegalHold.hs

```diff
@@ -274,3 +280,522 @@ testLHDeleteClientManually = do
+              & asList
+              >>= filterM \val -> (== "legalhold") <$> (val %. "type" & asString)
+          lhd %. "id" & asString
+
+      legalholdUserStatus tid alice bob `bindResponse` \resp -> do
+        resp.status `shouldMatchInt` 200
+        resp.json %. "client.id" `shouldMatch` lhdId
+        resp.json %. "status" `shouldMatch` "enabled"
+
+      replicateM 2 do
+        objId $ addClient bob def `bindResponse` getJSON 201
+        >>= traverse_ \client ->
+          awaitNotification bob client noValue isUserClientAddNotif >>= \notif -> do
+            notif %. "payload.0.client.type" `shouldMatch` "legalhold"
+            notif %. "payload.0.client.class" `shouldMatch` "legalhold"
+
+      -- the other team members receive a notification about the
+      -- legalhold device being approved in their team
+      for_ [alice, charlie] \user -> do
+        client <- objId $ addClient user def `bindResponse` getJSON 201
+        awaitNotification user client noValue isUserLegalholdEnabledNotif >>= \notif -> do
+          notif %. "payload.0.id" `shouldMatch` objId bob
+      for_ [ollie, sandy] \outsider -> do
+        outsiderClient <- objId $ addClient outsider def `bindResponse` getJSON 201
+        assertNoNotifications outsider outsiderClient Nothing isUserLegalholdEnabledNotif
+
+testLHGetDeviceStatus :: App ()
+testLHGetDeviceStatus =
+  startDynamicBackends [mempty] \[dom] -> do
+    -- team users
+    -- alice (team owner) and bob (member)
+    (alice, tid, [bob]) <- createTeam dom 2
+    for_ [alice, bob] \user -> do
+      legalholdUserStatus tid alice user `bindResponse` \resp -> do
+        resp.status `shouldMatchInt` 200
+        resp.json %. "status" `shouldMatch` "no_consent"
+
+    lpk <- getLastPrekey
+    pks <- replicateM 3 getPrekey
+
+    withMockServer (lhMockApp' (Just (lpk, pks))) \lhPort _chan -> do
+      legalholdWhitelistTeam tid alice
+        >>= assertStatus 200
+
+      legalholdUserStatus tid alice bob `bindResponse` \resp -> do
+        resp.status `shouldMatchInt` 200
+        resp.json %. "status" `shouldMatch` "disabled"
+        lookupField resp.json "last_prekey"
+          >>= assertNothing
+        runMaybeT (lookupFieldM resp.json "client" >>= flip lookupFieldM "id")
+          >>= assertNothing
+
+      -- the status messages for these have already been tested
+      postLegalHoldSettings tid alice (mkLegalHoldSettings lhPort)
+        >>= assertStatus 201
+
+      requestLegalHoldDevice tid alice bob
+        >>= assertStatus 201
+
+      approveLegalHoldDevice tid bob defPassword
+        >>= assertStatus 200
+
+      legalholdUserStatus tid alice bob `bindResponse` \resp -> do
+        resp.status `shouldMatchInt` 200
+        resp.json %. "status" `shouldMatch` "enabled"
+        resp.json %. "last_prekey" `shouldMatch` lpk
+      -- TODO(mangoiv): where do we take the LH device client
+      -- id from??
+      -- resp.json %. "client.id" `shouldMatch` _
```

Comment:
```suggestion
      -- We don't have the LH client id handy, so we have nothing to match `resp.json %. "client.id"` against.
```

Reply:
We can actually get these I think, I think I just missed to fix this, sorry.